### PR TITLE
[vm-validator] Update VM validator to return a VMValidatorResult instead of VMStatus

### PR DESCRIPTION
**Pull request by tzakian**

### CODE REVIEW

File: language/libra-vm/src/libra_vm.rs [hunk 1]
```diff
@@ -632,12 +632,13 @@ impl VMVerifier for LibraVM {
         &self,
         transaction: SignedTransaction,
         state_view: &dyn StateView,
-    ) -> Option<VMStatus> {
+    ) -> VMValidatorResult {
         let data_cache = BlockDataCache::new(state_view);
+        let gas_price = transaction.gas_unit_price();
```
Comment (sblackshear) [inline language/libra-vm/src/libra_vm.rs]: - TODO for normalization?
- Will this eventually be returned from the prologue + read from the return value of `verify_transaction_impl` below?

File: types/src/transaction/mod.rs [hunk 2]
```diff
@@ -635,6 +635,27 @@ impl From<VMStatus> for TransactionStatus {
     }
 }
 
+/// The result of running the transaction through the VM validator.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct VMValidatorResult {
+    status: Option<VMStatus>,
+    gas_price: u64,
```
Comment (sblackshear) [inline types/src/transaction/mod.rs]: Should we call this something like `normalized_gas_price` or `priority` just to emphasize that it is (or will be) different from the gas price on the `RawTransaction`?

File: types/src/transaction/mod.rs [hunk 3]
```diff
@@ -635,6 +635,27 @@ impl From<VMStatus> for TransactionStatus {
     }
 }
 
+/// The result of running the transaction through the VM validator.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct VMValidatorResult {
+    status: Option<VMStatus>,
```
Comment (sblackshear) [inline types/src/transaction/mod.rs]: If this is `None`, will the mempool ever read `gas_price`?

Comment (phoenix-antigravity) [reply types/src/transaction/mod.rs]: yes, mempool will use gas priority value. None just means there's no vm error

**Pull request closed without merging**